types/target.go

```diff
@@ -49,7 +49,16 @@ type TargetConfig struct {
 }
 
 func (tc *TargetConfig) String() string {
-	b, err := json.Marshal(tc)
+	var redacted_tc *TargetConfig
```

Comment:
I wonder would the effect be the same if you do

if tc.password != nil {
password := "****"
tc.password = &password
}